File: backend/src/main/java/io/github/daegwonkim/backend/common/config/CoolsmsConfig.java
package io.github.daegwonkim.backend.common.config;

import net.nurigo.sdk.NurigoApp;
import net.nurigo.sdk.message.service.DefaultMessageService;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class CoolsmsConfig {

    @Bean
    public DefaultMessageService defaultMessageService(
            @Value("${coolsms.url}") String coolsmsUrl,
            @Value("${coolsms.apiKey}") String coolsmsApiKey,
            @Value("${coolsms.secretKey}") String coolsmsSecretKey
    ) {
        return NurigoApp.INSTANCE.initialize(
                coolsmsApiKey,
                coolsmsSecretKey,
                coolsmsUrl
        );
    }
}

File: backend/src/main/kotlin/io/github/daegwonkim/backend/common/config/CoolsmsConfig.kt
package io.github.daegwonkim.backend.common.config

import io.github.daegwonkim.backend.common.properties.CoolsmsProperties
import net.nurigo.sdk.NurigoApp
import net.nurigo.sdk.message.service.DefaultMessageService
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
@EnableConfigurationProperties(CoolsmsProperties::class)
class CoolsmsConfig {

    @Bean
    fun defaultMessageService(
        properties: CoolsmsProperties
    ): DefaultMessageService = NurigoApp.initialize(
        properties.apiKey,
        properties.secretKey,
        properties.url
    )
}